.buildkite/publish/dra/init_dra_publishing.sh

```diff
@@ -0,0 +1,153 @@
+#!/bin/bash
+
+########
+# publishes a DRA artifact
+########
+
+set -exu
+set -o pipefail
+
+# Load our common environment variables for publishing
+export DRA_DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+CURDIR="$(dirname "$DRA_DIR")"
+source $CURDIR/publish-common.sh
+export RELEASE_DIR="${PROJECT_ROOT}"
+
+# Create and stage the artifact
+cd $PROJECT_ROOT
+make clean sdist
+mkdir -p $RELEASE_DIR/dist/dra-artifacts
+cd -
+
+echo "The artifacts are: $(ls $RELEASE_DIR/dist)"
+
+
+# ensure JQ is installed...
+if ! which jq > /dev/null; then
+  wget -O jq https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64 && chmod +x jq
+fi
+
+# Need to export the revision, name and version and others
+# for the dependency report and publishing scripts
+export REVISION="${BUILDKITE_COMMIT}"
+export BRANCH_NAME="${BUILDKITE_BRANCH}"
+export PRODUCT_NAME="connectors"
+export GIT_REPO="connectors"
+
+# set PUBLISH_SNAPSHOT and PUBLISH_STAGING based on the branch
+if [[ "${BUILDKITE_BRANCH:-}" =~ (main|[0-9]\.[0-9x]*$) ]]; then
+  export PUBLISH_SNAPSHOT="true"
+fi
+if [[ "${BUILDKITE_BRANCH:-}" =~ ([0-9]\.[0-9x]*$) ]]; then
+  export PUBLISH_STAGING="true"
+fi
+
+# Sanity check in the logs to list the downloaded artifacts
+chmod -R a+rw "${RELEASE_DIR}/dist"
+echo "---- listing artifact directory: ${RELEASE_DIR}/dist:"
+ls -al $RELEASE_DIR/dist
+
+# ensure we have a directory and permissions for our dependency report to be created
+# this path is set in the actual config file in
+# https://github.com/elastic/infra/tree/master/cd/release/release-manager/project-configs
+export DEPENDENCIES_REPORTS_DIR="${RELEASE_DIR}/dist/cd/output" # TODO, change this path?
+mkdir -p "${DEPENDENCIES_REPORTS_DIR}"
+chmod -R a+rw "${DEPENDENCIES_REPORTS_DIR}"
+
+# function to set our vault creds for DRA publishing
+function setDraVaultCredentials() {
+  set +x
+  echo "-- setting DRA vault credentials --"
+
+  # need to keep original vault addr to reset it
+  # after we're done or else it won't be set
+  # if we call this fxn again
+  export ORIGINAL_VAULT_ADDR="${VAULT_ADDR:-}"
+
+  DRA_CREDS=$(vault kv get -field=data -format=json kv/ci-shared/release/dra-role)
+  if [[ "${DRA_CREDS:-}" == "" ]]; then
+    echo "Could not get DRA credentials from vault"
+    exit 2
+  fi
+
+  export VAULT_ADDR="$(echo $DRA_CREDS | jq -r '.vault_addr')"
+  if [[ "${VAULT_ADDR:-}" == "" ]]; then
+    echo "Could not get VAULT_ADDR from DRA credentials"
+    exit 2
+  fi
+
+  export VAULT_ROLE_ID="$(echo $DRA_CREDS | jq -r '.role_id')"
+  if [[ "${VAULT_ROLE_ID:-}" == "" ]]; then
+    echo "Could not get VAULT_ADDR from DRA credentials"
+    exit 2
+  fi
+
+  export VAULT_SECRET_ID="$(echo $DRA_CREDS | jq -r '.secret_id')"
+  if [[ "${VAULT_SECRET_ID:-}" == "" ]]; then
+    echo "Could not get VAULT_ADDR from DRA credentials"
+    exit 2
+  fi
+
+  unset DRA_CREDS
```

Comment:
should we use `trap` to unset this if process exits or is interrupted?

Reply:
I don't think we need to worry, because if the process exits, nothing's going to log this env-var. I think this is here to make sure that nothing after this accidentally dumps the creds to buildkite logs.